bofire/strategies/strategy.py

```diff
@@ -24,12 +24,12 @@ def __init__(
     ):
         self.domain = data_model.domain
         # if data_model.seed is None (no explicit seed provided by the user),
-        # we draw a random seed from the default random number generator
-        # This is done to ensure reproducibility of the strategy,
+        # we use the randomly generated entropy from the seed sequence.
+        # This is done to ensure reproducibility of the strategy:
         # even if the user does not provide a seed one can extract the used seed
         # from the strategy object via `strategy.seed`.
-        self.seed = data_model.seed or np.random.default_rng().integers(1000)
-        self.rng = np.random.default_rng(self.seed)
+        self.seed_seq = np.random.SeedSequence(data_model.seed)
+        self.seed = self.seed_seq.entropy
```

Comment:
Let us make `self.seed` to a property to prevent a user to overwrite it:

```
@property
def seed(self) --> int:
    return self.seed_seq.entropy
```